### PR TITLE
Added unit tests for /pkg/version

### PR DESCRIPTION
**Pull request by Pushkarm029**
## Which problem is this PR solving?
- Part of #5068 

## Description of the changes
- added some missing unit tests in /pkg/version

## How was this change tested?
- go test

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: pkg/version/command_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	cmd := Command()
+
+	expectedUse := "version"
+	assert.Equal(t, expectedUse, cmd.Use, "Command use should be '%s'", expectedUse)
+
+	expectedShortDescription := "Print the version."
+	assert.Equal(t, expectedShortDescription, cmd.Short, "Command short description should be '%s'", expectedShortDescription)
+
+	expectedLongDescription := `Print the version and build information.`
+	assert.Equal(t, expectedLongDescription, cmd.Long, "Command long description should be '%s'", expectedLongDescription)
```
Comment (yurishkuro) [inline pkg/version/command_test.go:38]: these seem unnecessary

File: pkg/version/command_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	var b bytes.Buffer
+	cmd.SetOut(&b)
+	cmd.Execute()
+	out, err := io.ReadAll(&b)
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (yurishkuro) [inline pkg/version/command_test.go:46]:
```suggestion
	require.NoError(t, err)
```

File: pkg/version/command_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	expectedShortDescription := "Print the version."
+	assert.Equal(t, expectedShortDescription, cmd.Short, "Command short description should be '%s'", expectedShortDescription)
+
+	expectedLongDescription := `Print the version and build information.`
+	assert.Equal(t, expectedLongDescription, cmd.Long, "Command long description should be '%s'", expectedLongDescription)
+
+	var b bytes.Buffer
+	cmd.SetOut(&b)
+	cmd.Execute()
```
Comment (yurishkuro) [inline pkg/version/command_test.go:42]: if this returns error, then add `require.NoError(t, ..)`

File: pkg/version/build_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+func TestGet(t *testing.T) {
+	commitSHA = "foobar"
+	latestVersion = "v1.2.3"
+	date = "2024-01-04"
+
+	info := Get()
+
+	assert.Equal(t, commitSHA, info.GitCommit, "Command long description should be '%s'", commitSHA)
+	assert.Equal(t, latestVersion, info.GitVersion, "Command long description should be '%s'", latestVersion)
+	assert.Equal(t, date, info.BuildDate, "Command long description should be '%s'", date)
```
Comment (yurishkuro) [inline pkg/version/build_test.go:32]: Throughout the PR the messages are redundant, the output from assert will be already informative enough
```suggestion
	assert.Equal(t, date, info.BuildDate)
```

File: pkg/version/handler_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	commitSHA = "foobar"
+	latestVersion = "v1.2.3"
+	date = "2024-01-04"
+	expectedJSON := []byte(`{"gitCommit":"foobar","gitVersion":"v1.2.3","buildDate":"2024-01-04"}`) // Replace with expected values
+
+	mockLogger := zap.NewNop()
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		RegisterHandler(http.DefaultServeMux, mockLogger)
+		http.DefaultServeMux.ServeHTTP(w, r)
```
Comment (yurishkuro) [inline pkg/version/handler_test.go:38]: you don't register handlers inside a handler. I think you want something like
```
mux := http.NewMux()
RegisterHandler(mux, logger)
s := httptest.NewServer(mux)
```

File: pkg/version/handler_test.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	}))
+	defer server.Close()
+
+	resp, err := http.Get(server.URL + "/version")
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (yurishkuro) [inline pkg/version/handler_test.go:45]: use require.NoError

File: pkg/version/handler_test.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("Expected status code 200, got %d", resp.StatusCode)
+	}
```
Comment (yurishkuro) [inline pkg/version/handler_test.go:49]: use asserts for expectations

**Comment by Pushkarm029** [comment]
@yurishkuro resolved issues✅

File: pkg/version/handler_test.go [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	server := httptest.NewServer(mux)
+
+	defer server.Close()
+
+	resp, err := http.Get(server.URL + "/version")
+	require.NoError(t, err)
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	body, err := io.ReadAll(resp.Body)
+	resp.Body.Close()
+	require.NoError(t, err)
```
Comment (yurishkuro) [inline pkg/version/handler_test.go:46]: these are in the wrong order

**Review by yurishkuro — COMMENTED** [review]
please remove the change to the IDL submodule:
```
git checkout main
git pull
git submodule update
git checkout your-branch
git commit -sm 'fix idl'
```

**Comment by Pushkarm029** [comment]
@yurishkuro fixed

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**